src/main/java/net/minecraftforge/fmp/block/BlockCoverable.java

```diff
@@ -0,0 +1,730 @@
+        if (container == null) { return getWeakPowerDefault(state, world, pos, side, null); }
+        return Math.max(container.getPartContainer().getWeakSignal(side), getWeakPowerDefault(state, world, pos, side, container));
+    }
+
+    public int getWeakPowerDefault(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side, MicroblockContainer partContainer)
+    {
+        return super.getWeakPower(state, world, pos, side);
+    }
+
+    @Override
+    public final int getStrongPower(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side)
+    {
+        if (side == null) { return 0; }
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container == null) { return getStrongPowerDefault(state, world, pos, side, null); }
+        return Math.max(container.getPartContainer().getStrongSignal(side), getStrongPowerDefault(state, world, pos, side, container));
+    }
+
+    public int getStrongPowerDefault(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side,
+            MicroblockContainer partContainer)
+    {
+        return super.getStrongPower(state, world, pos, side);
+    }
+
+    @Override
+    public final boolean isSideSolid(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container == null) { return false; }
+        return container.getPartContainer().isSideSolid(side) || isSideSolidDefault(state, world, pos, side);
+    }
+
+    public boolean isSideSolidDefault(IBlockState state, IBlockAccess world, BlockPos pos, EnumFacing side)
+    {
+        return super.isSideSolid(state, world, pos, side);
+    }
+
+    @Override
+    public final boolean canPlaceTorchOnTop(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container == null) { return false; }
+        return container.getPartContainer().canPlaceTorchOnTop() || canPlaceTorchOnTopDefault(state, world, pos);
+    }
+
+    public boolean canPlaceTorchOnTopDefault(IBlockState state, IBlockAccess world, BlockPos pos)
+    {
+        return super.canPlaceTorchOnTop(state, world, pos);
+    }
+
+    @Override
+    @SideOnly(Side.CLIENT)
+    public final void randomDisplayTick(IBlockState state, World world, BlockPos pos, Random rand)
+    {
+        randomDisplayTickDefault(state, world, pos, rand);
+        IMicroblockContainerTile tile = getMicroblockTile(world, pos);
+        MicroblockContainer container = tile != null ? tile.getMicroblockContainer() : null;
+        if (container != null)
```

Comment:
While asie is right in saying that you are supposed to implement your own implementations for the methods in IMicroblockContainerTile, I agree that it should be turned into a capability. It will not have a default implementation, though, unless you want me to be hacky and check microblock placement against the original block's collision boxes.